harris/harris.go

```diff
@@ -0,0 +1,195 @@
+package harris
+
+import (
+	"yi/matrix"
+	"image"
+	"yi/point"
+	"log"
+)
+const k = 0.04
+
+func smoothFilter() matrix.Matrix {
+	result := matrix.Ones(3, 3).Scale(float64(1)/float64(9))
+	return result
+}
+
+func gaussianFilter() matrix.Matrix {
+	arr := [][]float64 {
+		[]float64{1, 4, 7, 4, 1},
+		[]float64{4, 16, 26, 16, 4},
+		[]float64{7, 26, 41, 26, 7},
+		[]float64{4, 16, 26, 16, 4},
+		[]float64{1, 4, 7, 4, 1},
+	}
+	newMatrix := matrix.Matrix(arr)
+	return newMatrix.Scale(float64(1)/newMatrix.Sum())
+}
+
+func imageGradientX() matrix.Matrix {
+	arr := [][]float64 {
+		[]float64{ -1, 0, 1},
+		[]float64{ -1, 0, 1},
+		[]float64{ -1, 0, 1},
+	}
+	return matrix.Matrix(arr)
+}
+
+func imageGradientY() matrix.Matrix {
+	return imageGradientX().Ctranspose()
+}
+
+func isLargestNeighbor(m matrix.Matrix, xPos int, yPos int) bool {
+	row := m.Row()
+	col := m.Col()
+
+	for i := xPos - 1; i <= xPos + 1; i++ {
+		for j := yPos - 1; j <= yPos + 1; j++ {
+			if i >= 0 && j >= 0 && i < row && j < col {
+				if m[xPos][yPos] < m[i][j] {
+					return false
+				}
+			}
+		}
+	}
+
+	return true
+}
+
+// compute the image gradient
+func imageGradient(img matrix.Matrix) (matrix.Matrix, matrix.Matrix, error) {
+	imgX, err := img.Filter(imageGradientX())
+	if err != nil {
+		log.Println("Error Filter image gradient x")
```

Comment:
I think you're probably ok here, but if this were production code we normally wouldn't have the `log` here since we're returning the error. We might decorate the error with extra context using `pkg/error` though.